app/helpers/ExerciseConfig/Pipeline/Box/Boxes/JvmRunBox.php

```diff
@@ -0,0 +1,151 @@
+                new Port(
+                    (new PortMeta())->setName(self::$INPUT_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port(
+                    (new PortMeta())->setName(self::$CLASS_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port((new PortMeta())->setName(self::$JAR_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)),
+                new Port((new PortMeta())->setName(self::$RUNNER_EXEC_PORT_KEY)->setType(VariableTypes::$STRING_TYPE)),
+                new Port((new PortMeta())->setName(self::$CLASSPATH_PORT_KEY)->setType(VariableTypes::$STRING_ARRAY_TYPE))
+            );
+            self::$defaultOutputPorts = array(
+                new Port((new PortMeta())->setName(self::$STDOUT_FILE_PORT_KEY)->setType(VariableTypes::$FILE_TYPE)),
+                new Port((new PortMeta())->setName(self::$OUTPUT_FILE_PORT_KEY)->setType(VariableTypes::$FILE_TYPE))
+            );
+        }
+    }
+
+    /**
+     * ElfExecutionBox constructor.
+     * @param BoxMeta $meta
+     */
+    public function __construct(BoxMeta $meta)
+    {
+        parent::__construct($meta);
+    }
+
+
+    /**
+     * Get type of this box.
+     * @return string
+     */
+    public function getType(): string
+    {
+        return self::$BOX_TYPE;
+    }
+
+    /**
+     * Get default input ports for this box.
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function getDefaultInputPorts(): array
+    {
+        self::init();
+        return self::$defaultInputPorts;
+    }
+
+    /**
+     * Get default output ports for this box.
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function getDefaultOutputPorts(): array
+    {
+        self::init();
+        return self::$defaultOutputPorts;
+    }
+
+    /**
+     * Get default name of this box.
+     * @return string
+     */
+    public function getDefaultName(): string
+    {
+        return self::$DEFAULT_NAME;
+    }
+
+    /**
+     * Compile box into set of low-level tasks.
+     * @param CompilationParams $params
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function compile(CompilationParams $params): array
+    {
+        $task = $this->compileBaseTask($params);
+        $task->setCommandBinary($this->getInputPortValue(self::$RUNNER_EXEC_PORT_KEY)->getValue());
+
+        // well we are running java and java is not smart enough to derive class
+        // name from class filename, so we are gonna be nice and do this tedious job
+        // instead of java runtime, you are welcome
+        $runnerClass = $this->getInputPortValue(self::$RUNNER_FILE_PORT_KEY)->getValue();
+        if (Strings::endsWith($runnerClass, ".class")) {
+            $runnerLength = Strings::length($runnerClass);
+            $runnerClass = Strings::substring($runnerClass, 0, $runnerLength - 6);
```

Comment:
At least write a comment that 6 is the length of '.class' (I know that now, but lets make sure I will not for get when I grow senile...)

Reply:
Well, I thought it is quite self-explanatory, that two lines above there is check for `endsWith`... but if you insist...